pennylane/transforms/decompose.py

```diff
@@ -655,6 +665,18 @@ def circuit():
             translate_op_alias(gate) for gate in gate_set if isinstance(gate, str)
         )
 
+    if isinstance(gate_set, dict):
```

Comment:
There seems to be a lot of duplicate logic here. While you're at it, would you mind abstracting this into a helper function called `_resolve_gate_set` so that both the `decompose` transform and `DecomposeInterpreter` can use it?